mdn/src/main/scala/org/apache/james/mdn/MDNReportParser.scala

```diff
@@ -0,0 +1,475 @@
+                         %d59-126           ;  characters not including
+                                            ;  ":".   */
+  private def ftext: Rule0 = rule {
+    CharPredicate(33.toChar to 57.toChar) |
+    CharPredicate(59.toChar to 126.toChar)
+  }
+
+  //   CFWS            =   (1*([FWS] comment) [FWS]) / FWS
+  private def cfws: Rule0 = rule { (oneOrMore(optional(fws) ~ comment) ~ fws) | fws }
+
+  //   FWS             =   ([*WSP CRLF] 1*WSP) /  obs-FWS
+  private def fws: Rule0 = rule { (optional(zeroOrMore(wsp) ~ crlf) ~ oneOrMore(wsp)) | obsFWS }
+
+  //         WSP            =  SP / HTAB
+  private def wsp: Rule0 = rule { sp | htab }
+
+  //         SP             =  %x20
+  private def sp: Rule0 = rule { ch(0x20) }
+
+  //         HTAB           =  %x09
+  private def htab: Rule0 = rule { ch(0x09) }
+
+  //         CRLF           =  CR LF
+  private def crlf: Rule0 = rule { cr ~ lf }
+
+  //         CR             =  %x0D
+  private def cr: Rule0 = rule { ch(0x0d) }
+
+  //         LF             =  %x0A
+  private def lf: Rule0 = rule { ch(0x0a) }
+
+  //   obs-FWS         =   1*WSP *(CRLF 1*WSP)
+  private def obsFWS: Rule0 = rule { oneOrMore(wsp) ~ zeroOrMore(crlf ~ oneOrMore(wsp)) }
+
+  //   comment         =   "(" *([FWS] ccontent) [FWS] ")"
+  private def comment: Rule[HNil, HNil] = rule { "(" ~ zeroOrMore(optional(fws) ~ ccontent) ~ optional(fws) ~ ")" }
+
+  //   ccontent        =   ctext / quoted-pair / comment
+  private def ccontent: Rule[HNil, HNil] = rule { ctext | quotedPair | comment }
+
+  /*   ctext           =   %d33-39 /          ; Printable US-ASCII
+                         %d42-91 /          ;  characters not including
+                         %d93-126 /         ;  "(", ")", or "\"
+                         obs-ctext   */
+  private def ctext = rule {
+    CharPredicate(33.toChar to 39.toChar) |
+    CharPredicate(42.toChar to 91.toChar) |
+    CharPredicate(93.toChar to 126.toChar) |
+    obsCText
+  }
+
+  //   obs-ctext       =   obs-NO-WS-CTL
+  private def obsCText = rule { obsNoWsCtl }
+
+  /*   obs-NO-WS-CTL   =   %d1-8 /            ; US-ASCII control
+                         %d11 /             ;  characters that do not
+                         %d12 /             ;  include the carriage
+                         %d14-31 /          ;  return, line feed, and
+                         %d127              ;  white space characters   */
+  private def obsNoWsCtl = rule {
+    CharPredicate(33.toChar to 39.toChar) |
+    ch(11) |
+    ch(12) |
+    CharPredicate(14.toChar to 31.toChar) |
+    ch(127)
+  }
+
+  //   quoted-pair     =   ("\" (VCHAR / WSP)) / obs-qp
+  private def quotedPair: Rule0 = rule { ("\\" ~ (vchar | wsp)) | obsQp }
+
+  //         VCHAR          =  %x21-7E
+  private def vchar: Rule0 = rule { CharPredicate(21.toChar to 0x7e.toChar) }
+
+  //   obs-qp          =   "\" (%d0 / obs-NO-WS-CTL / LF / CR)
+  private def obsQp: Rule0 = rule { "\\" ~ (ch(0xd0) | obsCText | lf | cr) }
```

Comment:
not obsCText but obsNoWsCtl

Reply:
good catch :) was missed in JAMES-3200 too

Reply:
but it didn'd introduce bugs as :  
   obs-ctext       =   obs-NO-WS-CTL